mne/io/eeglab/_eeglab.py

```diff
@@ -71,13 +72,91 @@ def _check_for_scipy_mat_struct(data):  # taken from pymatreader.utils
     return data
 
 
-def _readmat(fname, uint16_codec=None):
+def _scipy_reader(file_name, variable_names=None, uint16_codec=None):
+    """Load with scipy and then run the check function."""
+    mat_data = loadmat(
+        file_name, squeeze_me=True, mat_dtype=False, uint16_codec=uint16_codec
+    )
+    return _check_for_scipy_mat_struct(mat_data)
+
+
+def _readmat(fname, uint16_codec=None, *, preload=False):
     try:
         read_mat = _import_pymatreader_funcs("EEGLAB I/O")
     except RuntimeError:  # pymatreader not installed
-        eeg = loadmat(
-            fname, squeeze_me=True, mat_dtype=False, uint16_codec=uint16_codec
+        read_mat = _scipy_reader
+
+    # First handle the preload=False case
+    if not preload:
+        # when preload is `False`, we need to be selective about what we load
+        # and handle the 'data' field specially
+
+        # the files in eeglab are always the same field names
+        # the the fields were taken from the eeglab sample reference
+        # available at the eeglab github:
+        # https://github.com/sccn/eeglab/blob/develop/sample_data/eeglab_data.set
+        # The sample reference is the big reference for the field names
+        # in eeglab files, and what is used in the eeglab tests.
+        info_fields = """
+            setname filename filepath subject group condition session comments
+            nbchan trials pnts srate xmin xmax times icaact icawinv icasphere
+            icaweights icachansind chanlocs urchanlocs chaninfo ref event
+            urevent eventdescription epoch epochdescription reject stats
+            specdata specicaact splinefile icasplinefile dipfit history saved
+            etc
+        """.split()
+
+        # We first load only the info fields that are not data
+        # Then we check if 'data' is present and load it separately if needed
+        mat_data = read_mat(
+            fname,
+            variable_names=info_fields,
+            uint16_codec=uint16_codec,
         )
-        return _check_for_scipy_mat_struct(eeg)
-    else:
+
+        # checking the variables in the .set file
+        # to decide how to handle 'data' variable
+        try:
+            variables = whosmat(str(fname))
+        except NotImplementedError:
+            # TODO: find an alternative to whosmat that works with
+            # v7.3 matlab files that have h5py structure
+            preload = True
+            return read_mat(fname, uint16_codec=uint16_codec)
```

Comment:
This is confusing... this `preload=True` will have no effect?

Reply:
Yes, there are two cases there that I could not find a way to preload, one from scipy limitation in the whosmat from scipy that raises an error in case of a MATLAB file that has h5py, and the second case is when .set value has the values themselves, and I could not find them in the variables.

Reply:
For 7.3+ files the `.mat` should be readable with `h5py` directly. So you could do that here with a `with h5py.File(...)` etc.

If you don't feel like doing that now, I think it's okay to leave it as a TODO but you should remove the `preload=True` statement on line 123 -- it does nothing. You can explicit with `read_mat(...., preload=True)` on line 124 if you want